pkg/runner/step_context.go

```diff
@@ -237,6 +237,11 @@ func (sc *StepContext) setupShellCommand() common.Executor {
 		if step.Shell == "" {
 			step.Shell = rc.Run.Workflow.Defaults.Run.Shell
 		}
+		if rc.Run.Job().Container() != nil {
+			if rc.Run.Job().Container().Image != "" && step.Shell == "" {
+				step.Shell = "sh"
```

Comment:
Maybe the `Step` struct should init with `Step.Shell: "sh"` by default? Could simplify a few things down below?

Reply:
No, https://github.com/nektos/act/pull/853#issue-769923916